task/src/task.rs

```diff
@@ -480,13 +496,27 @@ impl<'pin, HSTT: HotShotTaskTypes> ProjectedHST<'pin, HSTT> {
                             let mut fut = handle_msg(msg, state);
                             match fut.as_mut().poll(cx) {
                                 Poll::Ready((result, state)) => {
-                                    *self.in_progress_fut = None;
-                                    *self.state = Some(state);
                                     if let Some(completed) = result {
                                         *self.r_val = Some(completed);
```

Comment:
Do we need to reset `self.in_progress_fut` and `self.state` here, as we do above?

Reply:
Yes! This is a bug. Pushing fix. We never actually use the state, which is why nothing broke.